test/ecto/repo_test.exs

```diff
@@ -894,4 +906,16 @@ defmodule Ecto.RepoTest do
       TestRepo.delete!(changeset)
     end
   end
+
+  test "handles nested embeds on delete" do
+    sub_embed = %SubEmbed{id: @uuid, y: "xyz"}
+    embed = %MyEmbed{id: @uuid, x: "xyz", sub_embed: sub_embed}
+    TestRepo.delete!(%MyModel{id: 1, embed: embed})
+    assert [{:after_delete, MyModel}, {:after_delete, MyEmbed},
+            {:after_delete, SubEmbed}, {:before_delete, SubEmbed},
+            {:before_delete, MyEmbed}, {:before_delete, MyModel} | _] =
+      Agent.get(CallbackAgent, &get_models/1)
+  end
+
+  # Nested assocs on delete are handled with the on_delete setting and tested there
```

Comment:
All previous tests come in pairs for embeds and assocs. This one doesn't. I wanted to make clear why. Is there a better place to put this?